backend/src/middleware/permissionsMiddleware.js

```diff
@@ -48,6 +54,13 @@ const belongsToMe = rule({
   return Boolean(notification)
 })
 
+/* TODO: decide if we want to remove this check: the check
+ * `onlyEnabledContent` throws authorization errors only if you have
+ * arguments for `disabled` or `deleted` assuming these are filter
+ * parameters. Soft-delete middleware obfuscates data on its way out
+ * anyways. Furthermore, `neo4j-graphql-js` offers many ways to filter for
+ * data so I believe, this is not a good check anyways.
+ */
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-14T09:39:52Z" title="Friday, June 14th 2019, 11:39:52 am +02:00">Jun 14, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -48,6 +54,13 @@ const belongsToMe = rule({
   return Boolean(notification)
 })
 
+/* TODO: decide if we want to remove this check: the check
+ * `onlyEnabledContent` throws authorization errors only if you have
+ * arguments for `disabled` or `deleted` assuming these are filter
+ * parameters. Soft-delete middleware obfuscates data on its way out
+ * anyways. Furthermore, `neo4j-graphql-js` offers many ways to filter for
+ * data so I believe, this is not a good check anyways.
+ */
```


What do we do about `enabledContent`? Shall I remove it?

Reply:
> [<img alt="ulfgebhardt" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-06-14T11:14:12Z" title="Friday, June 14th 2019, 1:14:12 pm +02:00">Jun 14, 2019</time>_

---


This clearly depends on who does the filtering - if its the server filtering stuff its fine to remove the check. If the filter is defined by the client and not enforced by the server, we need to make sure only content is delivered which the requesting user is allowed to see.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-14T11:21:32Z" title="Friday, June 14th 2019, 1:21:32 pm +02:00">Jun 14, 2019</time>_

---


The 100% security that the users see only what he is allowed to see is **not** handled by this check.

The check does sth. else: If you filter for posts that are deleted **only**, you will get an authentication error right away in case you are not an administrator. But if we remove this check, then you would get a result set with all deleted posts obfuscated. Same effect but different behaviour.

The obfuscation is the 100% security, because it's implemented on the post itself, not the top-level query. So even if you request nested posts (e.g. all posts of a user) you will see the obfuscated data.